client/src/components/pages/database/license-availability-table.tsx

```diff
@@ -10,9 +33,38 @@ const LicenseAvailabilityTable = () => {
     'Total Licenses'
   ]
 
+
+  
+  const formatData = (data: BusinessLicense[], zipcodeList: Set<EligibleBostonZipcode>) => {
+      const zips = [...zipcodeList]
+      const d = zips.map((zipcode) => {
+       const {totalAvailable, allAlcoholAvailable, beerWineAvailable} = getAvailableLicensesByZipcode(data, zipcode); 
+       const entry = {
+           rowData: [zipcode, String(totalAvailable), '-', String(MAX_AVAILABLE_PER_ZIP - totalAvailable), String(MAX_AVAILABLE_PER_ZIP)],
+           subRowData: [
+             ['All Alcohol Licenses', String(allAlcoholAvailable), '-', String(MAX_ALL_ALC_PER_ZIP - allAlcoholAvailable), String(MAX_ALL_ALC_PER_ZIP)],
+             ['Beer & Wine Licenses', String(beerWineAvailable), '-', String(MAX_BEER_WINE_PER_ZIP - beerWineAvailable), String(MAX_BEER_WINE_PER_ZIP)],
+           ] 
+       }
+
+       return entry as RowWithSubRows;
+
+      })
+
+      return d;
+  }
```

Comment:
Considering this function does not directly affect or depend on state, it should be declared outside of the react component. Above it in this file, or in another file would be fine. It's fairly small, so keeping it in this file may be good for readability, but I'lll leave that choice to you.

There is a tiny performance impact that comes with defining the function here. The declaration will happen every time the component is updated. Though the real benefits come from clearer separation of concerns and easier testing.
```suggestion
```